libbeat/monitoring/inputmon/input.go

```diff
@@ -81,7 +103,65 @@ func globalRegistry() *monitoring.Registry {
 }
 
 // MetricSnapshotJSON returns a snapshot of the input metric values from the
-// global 'dataset' monitoring namespace encoded as a JSON array (pretty formatted).
-func MetricSnapshotJSON() ([]byte, error) {
-	return json.MarshalIndent(filteredSnapshot(globalRegistry(), ""), "", "  ")
+// global 'dataset' monitoring namespace and from the localReg parameter
+// encoded as a JSON array (pretty formatted). It's safe to pass in a nil
+// localReg.
+func MetricSnapshotJSON(reg *monitoring.Registry) ([]byte, error) {
+	return json.MarshalIndent(filteredSnapshot(globalRegistry(), reg, ""), "", "  ")
+}
+
+// NewMetricsRegistry creates a monitoring.Registry for an input.
+//
+// The metric registry is created on parent with
+// name 'inputID' (all '.' are replaced by '_') and populated with
```

Comment:
To me this reads like the name is the string literal `inputID`. I suggest something like "The metric registry is created on the parent using inputID as the name where all '.' are replaced by '_'. The new registry is initialized with . . ."